Extensions/TileMap/tilemapruntimeobject.js

```diff
@@ -0,0 +1,273 @@
+
+gdjs.TileMapRuntimeObject.prototype.getRendererObject = function() {
+  return this._renderer.getRendererObject();
+};
+
+gdjs.TileMapRuntimeObject.prototype.update = function(runtimeScene) {
+  if (this._animationSpeedScale <= 0 || this._animationFps === 0) return;
+  var elapsedTime = this.getElapsedTime(runtimeScene) / 1000; 
+  
+  this._frameElapsedTime += elapsedTime * this._animationSpeedScale; 
+  while (this._frameElapsedTime > 1 / this._animationFps) {
+    this._renderer.incrementAnimationFrameX();
+    this._frameElapsedTime -= 1 / this._animationFps;
+  }
+  
+};
+/**
+ * @param { TileMapObjectDataType} oldObjectData
+ * @param { TileMapObjectDataType} newObjectData
+ */
+gdjs.TileMapRuntimeObject.prototype.updateFromObjectData = function(
+  oldObjectData,
+  newObjectData
+) {
+  if (oldObjectData.content.opacity !== newObjectData.content.opacity) {
+    this.setOpacity(newObjectData.content.opacity);
+  }
+  if (
+    oldObjectData.content.tilemapJsonFile !==
+    newObjectData.content.tilemapJsonFile
+  ) {
+    this.setTilemapJsonFile(newObjectData.content.tilemapJsonFile);
+  }
+  if (
+    oldObjectData.content.tilesetJsonFile !==
+    newObjectData.content.tilesetJsonFile
+  ) {
+    this.setTilesetJsonFile(newObjectData.content.tilesetJsonFile);
+  }
+  if (
+    oldObjectData.content.tilemapAtlasImage !==
+    newObjectData.content.tilemapAtlasImage
+  ) {
+    this.setTilemapAtlasImage(newObjectData.content.tilemapAtlasImage);
+  }
+  if (oldObjectData.content.displayMode !== newObjectData.content.displayMode) {
+    this.setDisplayMode(newObjectData.content.displayMode);
+  }
+  if (oldObjectData.content.layerIndex !== newObjectData.content.layerIndex) {
+    this.setLayerIndex(newObjectData.content.layerIndex);
+  }
+  if (
+    oldObjectData.content.animationSpeedScale !==
+    newObjectData.content.animationSpeedScale
+  ) {
+    this.setAnimationSpeed(newObjectData.content.animationSpeedScale);
+  }
+  if (
+    oldObjectData.content.animationFps !== newObjectData.content.animationFps
+  ) {
+    this.setAnimationFps(newObjectData.content.animationFps);
+  }
+
+  return true;
+};
+
+/**
+ * Initialize the extra parameters that could be set for an instance.
+ * @private
+ */
+gdjs.TileMapRuntimeObject.prototype.extraInitializationFromInitialInstance = function(
+  initialInstanceData
+) {
+  if (initialInstanceData.customSize) {
+    this.setWidth(initialInstanceData.width);
+    this.setHeight(initialInstanceData.height);
+  }
+};
+
+/**
+ * Set the Tilemap json file to display.
+ */
+gdjs.TileMapRuntimeObject.prototype.setTilemapJsonFile = function(tilemapJsonFile) {
+  this._tilemapJsonFile = tilemapJsonFile;
+  this._renderer.updateTileMap();
+};
+
+gdjs.TileMapRuntimeObject.prototype.getTilemapJsonFile = function() {
+  return this._tilemapJsonFile;
+};
+
+gdjs.TileMapRuntimeObject.prototype.isTilemapJsonFileName = function(selectedTilemapJsonFile) {
```

Comment:
Be consistent in the naming: either `isTilemapJsonFile` or `isTilemapJsonFileName`, but in this case put "name" for the getter/setter too.

Reply:
I decided to get rid of the word 'Name' from all getters/setters

Reply:
Sounds good then 👍